### ISSUE TITLE
Pollen vs Org for notes, data, and publishing

### ISSUE THREAD

**Issue by katerinjo** [mbutterick/pollen-users#14]
**TLDR:** How well does Pollen handle use cases beyond just fancy typography? Am I better off doing everything in org mode that isn't a web page or book?

I've been using org mode for the last month or so for most of my note-taking and writing. It's nice for certain things and the tooling is better than I'd be able to put together myself in a short amount of time, but like other formats I've experimented with, I keep running into problems with the same few needs: flexibility/extensibility, legibility for humans, and legibility for computers.

For example, I want to create and publish a dictionary/glossary of in-group jargon for a couple communities I'm a part of. I want reading and writing to be easy, so relational databases, XML, SXML, HTML, and JSON files are out. For definitions and usage notes I'll need inline emphasis, links, and phonetic transcription at the very least, so using multi-line strings for longish text in TOML is out. I might want to mess around with corpus linguistics questions using that data, or at least be able to do simple things like checking for repeat entries, so it has to be easily legible to programs I write, which is a problem with anything that isn't an established format with libraries like XML or simple like S-expressions or similar.

Org mode can *sorta* do what I want. Those inline classes can be written with macros that create HTML tags. There's an Elisp API and Python libraries for reading org files and parsing the HTML tags. There's nesting, properties, tagging, and some other neat features. But all of this is awkward and hacky.

I'm hoping that Pollen will solve these problems, but I don't want to spend another month experimenting with a format I might end up dropping yet again without a sanity check first.

Everything I hear about Pollen focuses on typography, which is legitimately cool, but I'm wondering how well it can handle everything else. It would be nice to have a personal information management system that I can read, my programs can read, that has expressive text built-in, and that lets me just dump whatever parts of it I want onto a website with little hassle.

Will Pollen do it? Will Pollen do it if I write a reasonable amount of extra code to force it into shape? Is there some better format for my use case? Or should I just build my own format from scratch?

**Comment by lijunsong** [comment]
When it comes to publishing, the most powerful system gives you the ability to transform any text in your document with little markups. You can do that with Pollen but you'll need some efforts to build your own infra (define your own markups). I used both org-mode and pollen for publishing webpages, pollen is definitely the way to go because org-mode has its own inconsistency and limitations (for example, the way to include html header is so awkward) that previously drove me crazy. Pollen is minimal, has few rules, and most importantly it's consistent (when a system is consistent, it will be simple, though not necessary easy at the beginning)! There is no such a problem as parsing the text because all your text result in s-expr in Pollen.

It's a different story when it comes to taking notes (like taking notes for lectures). Note taking is a problem of human computer interaction. That's what org-mode excels because of Emacs is a good interface. Pollen has no such an infrastructure (even though I have been trying to [help](https://github.com/lijunsong/pollen-rock)) at the moment; Pollen is still new, and the community is small, so the fancy toolchain is not there yet. In the current world of Pollen, there is no existing interaction for you to use TAB to jump between table's columns (like what org-mode gives you for free), but you can define your own text transformation to reduce typing.

Given your description, I think Pollen is better for you because all issues you mentioned are balance between readability between human and computers, which is what Pollen excels. And pollen's flexibility and extensibility is superior to any other markups.

Try it out. After the first a few tutorial you'll have the answer whether Pollen is for you. At least I had my answer when I finished the first tutorial.

**Comment by mbutterick** [comment]
When I’ve heard from org-moders in the past, I’ve suggested that it should be possible to add a `pollen/org-mode` dialect akin to `pollen/markdown` that parses org-mode files to X-expressions (which can then be processed further by the usual tag functions). This seems like a possibly more fruitful option than “build my own format from scratch”. 

OTOH I’m not volunteering to write an org-mode parser. I don’t use org mode myself. More broadly, I haven’t considered the possibilities of Pollen as a “personal information management system” (though I wouldn’t discourage others from doing so)
